src/Shell/Task/MigrationDiffTask.php

```diff
@@ -0,0 +1,426 @@
+     *
+     * @var array
+     */
+    protected $currentSchema;
+
+    /**
+     * List of the tables that are commonly found in the dump schema and the current schema
+     *
+     * @var array
+     */
+    protected $commonTables;
+
+    protected $templateData = [];
+
+    /**
+     * {@inheritDoc}
+     */
+    public function bake($name)
+    {
+        $this->setup();
+
+        if (!$this->checkSync()) {
+            $this->error('Your migrations history is not in sync with your migrations files. ' .
+                'Make sure all your migrations have been migrated before baking a diff.');
+            return 1;
+        }
+
+        if (empty($this->migrationsFiles) && empty($this->migratedItems)) {
+            return $this->bakeSnapshot($name);
+        }
+
+        $collection = $this->getCollection($this->connection);
+        EventManager::instance()->on('Bake.initialize', function (Event $event) use ($collection) {
+            $event->subject->loadHelper('Migrations.Migration', [
+                'collection' => $collection
+            ]);
+        });
+
+        return parent::bake($name);
+    }
+
+    /**
+     * Sets up everything the baking process needs
+     *
+     * @return void
+     */
+    public function setup()
+    {
+        $this->migrationsPath = $this->getPath();
+        $this->migrationsFiles = glob($this->migrationsPath . '*.php');
+        $this->phinxTable = $this->getPhinxTable($this->plugin);
+
+        $connection = ConnectionManager::get($this->connection);
+        $this->tables = $connection->schemaCollection()->listTables();
+        $tableExists = in_array($this->phinxTable, $this->tables);
+
+        $migratedItems = [];
+        if ($tableExists) {
+            $query = $connection->newQuery();
+            $migratedItems = $query
+                ->select(['version'])
+                ->from($this->phinxTable)
+                ->order(['version DESC'])
+                ->execute()->fetchAll('assoc');
+        }
+
+        $this->migratedItems = $migratedItems;
+    }
+
+    /**
+     * Get a collection from a database
+     *
+     * @param string $connection Database connection name.
+     * @return \Cake\Database\Schema\Collection
+     */
+    public function getCollection($connection)
+    {
+        $connection = ConnectionManager::get($connection);
+        return $connection->schemaCollection();
+    }
+
+    public function templateData()
+    {
+        $this->dumpSchema = $this->getDumpSchema();
+        $this->currentSchema = $this->getCurrentSchema();
+        $this->commonTables = array_intersect_key($this->currentSchema, $this->dumpSchema);
+
+        $this->calculateDiff();
+
+        return [
+            'data' => $this->templateData,
+            'dumpSchema' => $this->dumpSchema,
+            'currentSchema' => $this->currentSchema,
+        ];
+    }
+
+    protected function calculateDiff()
```

Comment:
Yes, there are a lot of missing doc blocks. One of my tasks before merge will be to do a sweep and add them all.